lib/ansible/utils/deepishcopy.py

```diff
@@ -0,0 +1,34 @@
+# (c) 2018 Matt Martz <matt@sivel.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.module_utils import six
+
+
+def deepishcopy(v):
+    """Function to create a deep copy of simple data types
+
+    Designed to be used within the Ansible "engine" to improve performance
+    issues where ``copy.deepcopy`` was used previously.
+
+    This may not support all data types, and was designed to handle specific workloads.
```

Comment:
We should list the data types here (and perhaps in comments in the code as well).  Right now, for instance, I'm not sure if leaving out tuples and sets is an explicit feature or an oversight.  Should also include information about datatypes we're trying to avoid (for instance, will using MutableMapping and Sequence instead of dict and list include those data types and that's why we check on dict and list instead?)

We should also have some way to test this specific workload.  A high level refactor to how values are stored needs to be able to validate that it doesn't regress the issue that this is fixing.

Reply:
I can list data types here.  Right now, we are only handling `list` and `dict` on purpose.

In the specific use case we are solving we should never be tuples or sets in the areas where this is currently used.  e.g. data coming from module responses, that has been previously serialized with JSON.

Also, not sure how I can really test it.  Other than running the function through `timeit` and compare it to being faster than `deepcopy`.